torch/csrc/jit/ir.h

```diff
@@ -163,6 +163,10 @@ struct Value {
  public:
   Value* setType(TypePtr type);
   void inferTypeFrom(const at::Tensor& output) {
+    if (output.is_mkldnn()) {
+      setType(DimensionedTensorType::create(output));
```

Comment:
maybe leave a comment why we do Dimensioned and not Complete?

Reply:
Now that this is more than one line, let's move this to the cpp file. ir.h is included in > 100 compilation units.